homeassistant/components/withings/common.py

```diff
@@ -1030,6 +1030,11 @@ async def async_get_data_manager(
     if const.DATA_MANAGER not in config_entry_data:
         profile = config_entry.data.get(const.PROFILE)
 
+        webhook_url = "{}{}".format(
+            get_url(hass, allow_internal=False, prefer_cloud=True),
+            webhook.async_generate_path(config_entry.data[CONF_WEBHOOK_ID]),
+        )
```

Comment:
We should not use `get_url` to generate cloud or webhooks.

For a reference implementation see:

<https://github.com/home-assistant/core/blob/79096864ebe6f187326289a3356341fea4252449/homeassistant/components/toon/coordinator.py#L58-L71>

Notes:
- The cloud hook needs to be stored and re-used
- Please use f-strings (instead of `format()` string interpolation).

Reply:
Done! (Although I'd have expected this to be a bit more of a common requirement and something that would be happening more often.)

Unfortunately I don't have a setup where I can test this with Cloud enabled :/ any help would be welcome.